sql/modules/Reconciliation.sql

```diff
@@ -166,20 +213,28 @@ disable the block_change_when_approved trigger on cr_report.$$;
 DROP FUNCTION IF EXISTS reconciliation__get_cleared_balance(int);
 CREATE OR REPLACE FUNCTION reconciliation__get_cleared_balance(in_chart_id int,
    in_report_date date DEFAULT date_trunc('second', now()))
-RETURNS numeric AS
+RETURNS NUMERIC AS
 $$
-    SELECT sum(ac.amount) * CASE WHEN c.category in('A', 'E') THEN -1 ELSE 1 END
-        FROM account c
-        JOIN acc_trans ac ON (ac.chart_id = c.id)
-    JOIN (      SELECT id FROM ar WHERE approved
-          UNION SELECT id FROM ap WHERE approved
-          UNION SELECT id FROM gl WHERE approved
-          ) g ON g.id = ac.trans_id
-    WHERE c.id = $1 AND cleared
-      AND ac.approved IS true
-      AND ac.transdate <= in_report_date
-    GROUP BY c.id, c.category;
-$$ LANGUAGE sql;
+DECLARE
+        result NUMERIC;
+BEGIN
+    SELECT COALESCE(
+        (SELECT SUM(ac.amount) * CASE WHEN c.category in('A', 'E') THEN -1 ELSE 1 END
+            INTO result
+            FROM account c
+            JOIN acc_trans ac ON (ac.chart_id = c.id)
+        JOIN (      SELECT id FROM ar WHERE approved
+              UNION SELECT id FROM ap WHERE approved
+              UNION SELECT id FROM gl WHERE approved
+              ) g ON g.id = ac.trans_id
```

Comment:
Can be replaced by a single SELECT on the `transaction` table.

Reply:
Fixed